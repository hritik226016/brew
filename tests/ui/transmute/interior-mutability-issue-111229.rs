//@ run-pass
// https://github.com/rust-lang/rust/issues/111229

pub struct Foo(std::cell::UnsafeCell<usize>);
pub struct Bar([u8; 0]);

pub fn foo(f: &Bar) {
    unsafe {
        let f = std::mem::transmute::<&Bar, &Foo>(f);
        //~^ WARNING transmuting from a type without interior mutability to a type with interior mutability
        //~| HELP `Foo` has interior mutability
        *(f.0.get()) += 1;
    }
}

fn main() {}
