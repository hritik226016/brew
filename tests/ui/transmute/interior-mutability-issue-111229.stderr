warning: transmuting from a type without interior mutability to a type with interior mutability is undefined behaviour if you modify the content
  --> $DIR/interior-mutability-issue-111229.rs:9:17
   |
LL |         let f = std::mem::transmute::<&Bar, &Foo>(f);
   |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = help: `Foo` has interior mutability
   = note: `#[warn(transmute_to_interior_mutability)]` on by default

warning: 1 warning emitted

